java/integTest/src/test/java/glide/BaseClientTests.java

```diff
@@ -0,0 +1,416 @@
+import static glide.api.models.commands.SetOptions.TimeToLiveType.KEEP_EXISTING;
+import static glide.api.models.commands.SetOptions.TimeToLiveType.MILLISECONDS;
+import static glide.api.models.commands.SetOptions.TimeToLiveType.UNIX_SECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import glide.api.BaseClient;
+import glide.api.RedisClient;
+import glide.api.RedisClusterClient;
+import glide.api.models.commands.SetOptions;
+import glide.api.models.configuration.NodeAddress;
+import glide.api.models.configuration.RedisClientConfiguration;
+import glide.api.models.configuration.RedisClusterClientConfiguration;
+import glide.api.models.exceptions.RequestException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import lombok.SneakyThrows;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class BaseClientTests {
+
+    private static final String KEY_NAME = "key";
+    private static final String INITIAL_VALUE = "VALUE";
+    private static final String ANOTHER_VALUE = "VALUE2";
+
+    private static RedisClient standaloneClient = null;
+    private static RedisClusterClient clusterClient = null;
+
+    private static List<BaseClient> clients = new ArrayList<>();
+
+    @BeforeAll
+    @SneakyThrows
+    public static void init() {
+        standaloneClient =
+                RedisClient.CreateClient(
+                                RedisClientConfiguration.builder()
+                                        .address(
+                                                NodeAddress.builder().port(TestConfiguration.STANDALONE_PORTS[0]).build())
+                                        .build())
+                        .get(10, SECONDS);
+
+        clusterClient =
+                RedisClusterClient.CreateClient(
+                                RedisClusterClientConfiguration.builder()
+                                        .address(NodeAddress.builder().port(TestConfiguration.CLUSTER_PORTS[0]).build())
+                                        .requestTimeout(5000)
+                                        .build())
+                        .get(10, SECONDS);
+
+        clients.add(standaloneClient);
+        clients.add(clusterClient);
+    }
+
+    @AfterAll
+    @SneakyThrows
+    public static void teardown() {
+        standaloneClient.close();
+        clusterClient.close();
+    }
+
+    @Test
+    @SneakyThrows
+    public void ping() {
```

Comment:
they may ask you to make this change in a separate PR

Reply:
Probably that's what I am thinking.

Reply:
Good news.  I've moved all the ping, & get/set calls to their upstream PRs.  I ended up calling the test file SharedCommandTests - like they did for Node tests.  
* https://github.com/aws/glide-for-redis/pull/919/files#diff-5db9a3bd41d647cfdba521ea248ea07d7c78bc9aaa41a26bb923d4f92a3efdd2
* https://github.com/aws/glide-for-redis/pull/914/files#diff-5db9a3bd41d647cfdba521ea248ea07d7c78bc9aaa41a26bb923d4f92a3efdd2